repository/observation_respository_test.go

```diff
@@ -0,0 +1,224 @@
+package repository
+
+import (
+	"fmt"
+	"github.com/ONSdigital/dp-dimension-importer/mocks"
+	"github.com/ONSdigital/dp-dimension-importer/model"
+	bolt "github.com/johnnadratowski/golang-neo4j-bolt-driver"
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+func TestNewObservationRepository(t *testing.T) {
+
+	Convey("Given mock parameters that do not return errors", t, func() {
+
+		connMock := &mocks.NeoConnMock{}
+		neo4jCliMock := &mocks.Neo4jClientMock{}
+		connectionPool := &mocks.NeoDriverPoolMock{
+			OpenPoolFunc: func() (bolt.Conn, error) {
+				return connMock, nil
+			},
+		}
+
+		Convey("When NewObservationRepository is called", func() {
+
+			repo, err := NewObservationRepository(connectionPool, neo4jCliMock)
+
+			Convey("Then the expected repository is returned with no error", func() {
+
+				So(repo.conn, ShouldEqual, connMock)
+				So(repo.neo4j, ShouldEqual, neo4jCliMock)
+				So(err, ShouldBeNil)
+			})
+
+			Convey("And connectionPool.OpenPool is called 1 time", func() {
+
+				So(len(connectionPool.OpenPoolCalls()), ShouldEqual, 1)
+			})
+		})
+	})
+}
+
+func TestNewObservationRepository_ConnectionError(t *testing.T) {
+
+	Convey("Given a mock connection pool that returns an error", t, func() {
+
+		neo4jCliMock := &mocks.Neo4jClientMock{}
+		connectionPool := &mocks.NeoDriverPoolMock{
+			OpenPoolFunc: func() (bolt.Conn, error) {
+				return nil, errorMock
+			},
+		}
+
+		Convey("When NewObservationRepository is called", func() {
+
+			repo, err := NewObservationRepository(connectionPool, neo4jCliMock)
+
+			Convey("Then the expected error is returned", func() {
+
+				So(err.Error(), ShouldEqual, errors.Wrap(errorMock, "connPool.OpenPool returned an error").Error())
+				So(repo, ShouldBeNil)
+			})
+		})
+	})
+}
+
+func TestObservationRepository_CreateConstraint_NilInstance(t *testing.T) {
+
+	Convey("Given an empty instance", t, func() {
+
+		var instance *model.Instance
+
+		connMock := &mocks.NeoConnMock{
+			CloseFunc: func() error {
+				return nil
+			},
+		}
+
+		repo := ObservationRepository{conn: connMock}
+
+		Convey("When CreateConstraint is invoked", func() {
+
+			err := repo.CreateConstraint(instance)
+
+			Convey("Then the expected error is returned", func() {
+				So(err.Error(), ShouldEqual, errors.New("instance is required but was nil").Error())
```

Comment:
no need to wrap that string then unwrap it (just use the string)?

Reply:
As above. FYI when using `errors.Wrap` you need to compare using the err.Error() method as you will run into equality issues comparing your test error against the error wrapping another error.

Reply:
Done 👍